cli/generate_permissions.py

```diff
@@ -28,48 +28,77 @@
+            return self_number < other_number
+
+        def __le__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number <= other_number
+
+        def __gt__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number > other_number
+
+        def __ge__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number >= other_number
+
+
+    class OrganisationManagementLevel(CompareRightLevel, Enum):
         SUPERADMIN = "superadmin"
         CAN_MANAGE_USERS = "can_manage_users"
         CAN_MANAGE_ORGANISATION = "can_manage_organisation"
-
-        def __init__(self, oml: str):
-            super().__init__()
-            self.numbers = {
+        NO_RIGHT = "no_right"
+
+        def get_number(self, value: "OrganisationManagementLevel") -> int:
+            if not isinstance(value, self.__class__):
+                raise PermissionException(
+                    f"The comparison expect an {self.__class__}-type and no string!"
+                )
+            numbers = {
                 "superadmin": 3,
                 "can_manage_organisation": 2,
                 "can_manage_users": 1,
+                "no_right": 0,
             }
-            self.number: int = self.numbers.get(oml, 0)
-
-        def is_ok(self, user_oml: str) -> bool:
-            return self.numbers.get(user_oml, 0) >= self.number
-
+            return numbers.get(value, 0)
 
-    class CommitteeManagementLevel(str, Enum):
+    class CommitteeManagementLevel(CompareRightLevel, Enum):
         \""" 2nd Permission Type, implemented as User.committee_as_manager_ids \"""
 
-        MANAGER = "can_manage_committees"
-
-        def __init__(self, cml: str):
-            super().__init__()
-            self.numbers = {
-                "can_manage_committees": 1,
+        MANAGER = "can_manage"
+        NO_RIGHT = "no_right"
+
+        def get_number(self, value: "CommitteeManagementLevel") -> int:
+            if not isinstance(value, self.__class__):
+                raise PermissionException(
+                    f"The comparison expect an {self.__class__}-type and no string!"
+                )
+            numbers = {
+                "can_manage": 1,
+                "no_right": 0,
             }
-            self.number: int = self.numbers.get(cml, 0)
-
-        def is_ok(self, user_cml: str) -> bool:
-            return self.numbers.get(user_cml, 0) >= self.number
+            return numbers.get(value, 0)
 
 
     class Permission(str):
         \""" Marker class to use typing with permissions. \"""
 
         def __str__(self) -> str:
-                return self.value  # type: ignore
+            return self.value  # type: ignore
     """
```

Comment:
Also, the whole code block got very large and does not need to be generated. I would suggest to just move this to another file like `management_levels.py` fmro which it can then be imported everywhere else.

Reply:
done